Packs/ReversingLabs_Titanium_Cloud/Integrations/ReversingLabsTitaniumCloudv2/ReversingLabsTitaniumCloudv2.py

```diff
@@ -1521,6 +1522,717 @@ def url_analyses_feed_from_date_output(response, start_time):
+
+    results = CommandResults(
+        outputs_prefix="ReversingLabs",
+        outputs={"domain_report": response_json},
+        readable_output=markdown,
+        indicator=indicator
+    )
+
+    return results
+
+
+def domain_downloaded_files_command():
+    domain_ti = create_domain_ti_object()
+
+    domain = demisto.getArg("domain")
+    classification = demisto.getArg("classification")
+    limit = int(demisto.getArg("result_limit"))
+    per_page = int(demisto.getArg("results_per_page"))
+
+    try:
+        response = domain_ti.get_downloaded_files_aggregated(
+            domain=domain,
+            classification=classification,
+            results_per_page=per_page,
+            max_results=limit
+        )
+    except Exception as e:
+        return_error(str(e))
+
+    results = domain_downloaded_files_output(response=response, domain=domain)
+    return_results(results)
+
+
+def domain_downloaded_files_output(response, domain):
+    files_table = tableToMarkdown(
+        name="Downloaded files",
+        t=response
+    )
+
+    markdown = f"## ReversingLabs Files downloaded from domain {domain}\n {files_table}"
+
+    dbot_score = Common.DBotScore(
+        indicator=domain,
+        indicator_type=DBotScoreType.DOMAIN,
+        integration_name="ReversingLabs TitaniumCloud v2",
+        score=0,
+        reliability=RELIABILITY
+    )
+
+    indicator = Common.Domain(
+        domain=domain,
+        dbot_score=dbot_score
+    )
+
+    results = CommandResults(
+        outputs_prefix="ReversingLabs",
+        outputs={"domain_downloaded_files": response},
+        readable_output=markdown,
+        indicator=indicator
+    )
+
+    return results
+
+
+def domain_urls_command():
+    domain_ti = create_domain_ti_object()
+
+    domain = demisto.getArg("domain")
+    limit = int(demisto.getArg("result_limit"))
+    per_page = int(demisto.getArg("results_per_page"))
+
+    try:
+        response = domain_ti.urls_from_domain_aggregated(
+            domain=domain,
+            results_per_page=per_page,
+            max_results=limit
+        )
+    except Exception as e:
+        return_error(str(e))
+
+    results = domain_urls_output(response=response, domain=domain)
+    return_results(results)
+
+
+def domain_urls_output(response, domain):
+    urls_table = tableToMarkdown(
+        name="URL list",
+        t=response
+    )
+
+    markdown = f"## ReversingLabs URL-s associated with domain {domain}\n {urls_table}"
+
+    dbot_score = Common.DBotScore(
+        indicator=domain,
+        indicator_type=DBotScoreType.DOMAIN,
+        integration_name="ReversingLabs TitaniumCloud v2",
+        score=0,
+        reliability=RELIABILITY
+    )
+
+    indicator = Common.Domain(
+        domain=domain,
+        dbot_score=dbot_score
+    )
+
+    results = CommandResults(
+        outputs_prefix="ReversingLabs",
+        outputs={"domain_urls": response},
+        readable_output=markdown,
+        indicator=indicator
+    )
```

Comment:
The `reversinglabs-titaniumcloud-domain-urls` command returns a list of URL-s. The `result_limit` argument defines the maximum number of URL-s the user wants to put into the context data. The `results_per_page` defines into how many requests the maximum number of URL-s will be split. The `results_per_page` is important to the user only because they might want to reduce or increase the number of requests towards our API. During the command it all accumulates into one list and ends up together in the context data. The `results_per_page` is just a way of chunking the end data into separate requests during the execution of the command.

Reply:
And to answer your other question:
If a user fires this command with the same domain twice but each time with a different `result_limit` (because that's what matters here actually), we don't expect the new data to replace the old data for that same domain. It's fine for the data to be updated under the same domain key.